internal/lookout/repository/store.go

```diff
@@ -257,6 +261,23 @@ func (r *SQLJobStore) upsertContainers(k8sId string, exitCodes map[string]int32)
 	return upsert(r.db, jobRunContainerTable, []string{"run_id", "container_name"}, containerRecords)
 }
 
+func (r *SQLJobStore) upsertAnnotations(jobId string, annotations map[string]string) error {
+	// Note: annotations are stored without the prefix
+	var annotationRecords []goqu.Record
+	for key, value := range annotations {
+		if strings.HasPrefix(key, r.annotationPrefix) && len(key) > len(r.annotationPrefix) {
+			trimmedKey := key[len(r.annotationPrefix):]
+			annotationRecords = append(annotationRecords, goqu.Record{
+				"job_id": jobId,
+				"key":    trimmedKey,
+				"value":  value,
+			})
+		}
+	}
+
+	return upsert(r.db, annotationTable, []string{"job_id", "key"}, annotationRecords)
+}
+
 func (r *SQLJobStore) determineJobState() exp.CaseExpression {
```

Comment:
It's a bit confusing having a data store that filters and trims values. Ideally datastores wouldn't change the data. Could this logic be moved elsewhere? What is the cost of just ditching this logic and storing all annotations including prefix? @jankaspar may have views.

Reply:
All the annotations are saved in the job table already in json format, this table is intended for lookup by user specified key value pairs. For that saving just prefixed annotations is much easier. We want to for example create a dropdown with all keys in ui. Maybe the name of the table could be improved to reflect this, any suggestions?

Reply:
Maybe `user_annotations`?

Reply:
user_annotations?
user_annotations_lookup?
user_annotations_index?

Reply:
Lets go with `user_annotations_lookup` and rename the methods to match the idea.